.github/README.md

```diff
@@ -0,0 +1,137 @@
+          generators/gemmini/software/gemmini-rocc-tests
+        key: ${{ inputs.group-key }}-${{ github.ref }}-${{ github.sha }}
+    - name: run rtl build script if not cached
+      run: |
+        if [[ "${{ steps.rtl-build-id.outputs.cache-hit }}" != 'true' ]]; then
+          echo "Cache miss on ${{ inputs.group-key }}-${{ github.ref }}-${{ github.sha }}"
+          ./.github/scripts/${{ inputs.build-script }} ${{ inputs.group-key }} ${{ inputs.build-type }}
+        else
+          echo "cache hit do not prepare rtl"
+        fi
+      shell: bash
+```
+
+Our own composite actions are defined in the `.github/actions/<ActionName>/action.yml`
+
+.github/scripts directory
+-------------------
+
+This directory contains most the collateral for the Chipyard CI to work.
+The following is included in `.github/scripts/: directory
+
+    `build-toolchains.sh`        # build either riscv-tools or esp-tools
+    `create-hash.sh`             # create hashes of riscv-tools/esp-tools to use as hash keys
+    `do-rtl-build.sh`            # use verilator to build a sim executable (remotely)
+    `defaults.sh`                # default variables used
+    `check-commit.sh`            # check that submodule commits are valid
+    `build-extra-tests.sh`       # build default chipyard tests located in tests/
+    `clean-old-files.sh`         # clean up build server files
+    `do-fpga-rtl-build.sh`       # similar to `do-rtl-build` but using fpga/
+    `install-verilator.sh`       # install verilator on build server
+    `run-firesim-scala-tests.sh` # run firesim scala tests
+    `run-tests.sh                # run tests for a specific set of designs
+
+How things are set up for Chipyard
+---------------------------------
+
+The steps for CI to run are as follows.
+1. Build the toolchains in parallel (note: `esp-tools` is currently not used in the run).
+The docker image sets up the `PATH` and `RISCV` variable so that `riscv-tools` is the default (currently the `env.sh` script that is created at tool build is unused).
+2. Create the simulator binary.
+This requires the `riscv-tools` for `fesvr` and `verilator` to be able to build the binary.
+This stores all collateral for the tests (srcs, generated-srcs, sim binary, etc) to run "out of the gate" in the next job (make needs everything or else it will run again).
+3. Finally, run the desired tests.
+
+Other CI Setup
+--------------
+
+To get the CI to work correctly you need to create the following GH Repository Secrets
+
+| Secret | Value |
+| -------| ------------- |
+| BUILDSERVER | the hostname of the remote build server (likely be a millennium machine) |
+| BUILDUSER | the login to use on the build server |
+| BUILDDIR | the directory to use on the build server |
+| SERVERKEY | a private key to access the build server |
+
+The default.sh script defines the following,
+```bash
+CI_DIR = /path/to/where/you/want/to/store/remote/files
+```` 
+but in the future this should likely be a GH Secret too.
```

Comment:
Not true anymore?

Reply:
What is the difference between `BUILDDIR` and `CI_DIR`? To me they sound the same...

Reply:
I see what you did. Marking complete.